src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstringTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Text;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.UnitTests
+{
+    public class StringBuilderAppendShouldNotTakeSubstringTests : DiagnosticAnalyzerTestBase
+    {
+        [Fact]
+        public void TestFromProposingTicketFirstLine()
+        {
+            const string code = @"
+using System.Text;
+
+public class C
+{
+    public string Append(string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text.Substring(2));
+        return sb.ToString();
+    }
+}";
+            var expectedDiagnostic =
+                new DiagnosticResult(
+                        StringBuilderAppendShouldNotTakeSubstring.RuleReplaceOneParameter)
+                    .WithLocation("Test0.cs", 9, 9)
+                    .WithMessage(
+                        StringBuilderAppendShouldNotTakeSubstring.RuleReplaceOneParameter.MessageFormat.ToString());
+
+            VerifyCSharp(code, expectedDiagnostic);
+        }
+
+        [Fact]
+        public void TestFromProposingTicketSecondLine()
+        {
+            const string code = @"
+using System.Text;
+
+public class C
+{
+    public string Append(string text)
+    {
+        var sb = new StringBuilder();
+        sb.Append(text.Substring(0,6));
+        return sb.ToString();
+    }
+}";
+            var expectedDiagnostic = new DiagnosticResult(
+                    StringBuilderAppendShouldNotTakeSubstring.RuleIdTwoParameterId,
+                    DiagnosticHelpers.DefaultDiagnosticSeverity)
```

Comment:
:bulb: Use `new DiagnosticResult(StringBuilderAppendShouldNotTakeSubstring.RuleReplaceTwoParameter)` to get the benefits described in my previous comment.

Reply:
thanks.